packages/relay-runtime/store/observeFragmentExperimental.js

```diff
@@ -173,6 +170,67 @@ function observeSelector<TFragmentType: FragmentType, TData>(
   });
 }
 
+function observePluralSelector<
+  TFragmentType: FragmentType,
+  TData: Array<mixed>,
+>(
+  environment: IEnvironment,
+  fragmentNode: Fragment<TFragmentType, TData>,
+  fragmentSelector: PluralReaderSelector,
+): Observable<FragmentState<TData>> {
+  const snapshots = fragmentSelector.selectors.map(selector =>
+    environment.lookup(selector),
+  );
+
+  return Observable.create(sink => {
+    const states = snapshots.map((snapshot, index) =>
```

Comment:
Might be worth a comment up here explaining that `states` is mutable and its values will get updated in place overtime

Reply:
https://github.com/facebook/relay/pull/4862/commits/06903bfa7a0030c3d5569051b47479e968627bf8